### ISSUE TITLE
Movie Thumbnails not Updating

### ISSUE THREAD

**Issue by erzats** [jellyfin-archive/jellyfin-roku-legacy#735]
**Describe the bug**
I uploaded two videos to my Movies folder. The automatic metadata scan calls both of them the same even though they were two different movies and assigns thumbnails accordingly (which I think is strange and maybe should be a bug on its own). When I changed the name and image of the incorrect video, the name changed in the Roku app, but the image stayed the same as the incorrect one. Also, I changed the background image but that didn't change either. However, when I select it and am offered the option to play and can view details, that has the correct thumbnail.

**To Reproduce**
1. Copy a new movie to your Movies folder.
2. Allow it to automatically grab metadata and populate the thumbnail and background image.
3. Log in through the browser as an admin.
4. 'Edit Metadata' of the incorrect movie and remove the metadata pertaining to the wrongly applied movie title.
5. Change the title.
6. 'Edit Images' and change the thumbnail and background.
7. The Roku app should still have the old image on the main Movies page but when you select it you should see the correct image.

**Expected behavior**
After changing the image as an admin, the new image should be displayed in the Roku client.

**Comment by caj411** [comment]
I have the same issue. 
I updated metadata on several movies using the identify feature. I was able find the correct info and images and save it. From my web browser it appears properly, but the updates do not appear in my Roku app.  It doesn't appear to be aware of the changes in the Jellyfin database.  It did however pickup some new movies that were added, just not the metadata.

**Comment by ifben** [comment]
I can sort of confirm this issue--the movie details are updated correctly, but the poster image is incorrect in the media grid view. However, doing a full reboot on my TV fixed this, which I imagine cleared out everything written to `tmp`. I would think `tmp` would clear any time the app exited, but I was able to exit the Jellyfin app and launch other apps, return to the Jellyfin app, and still have incorrect images. It was only after the full reboot that they displayed correctly.

**Comment by neilsb** [comment]
The Jellyfin Roku app doesn't do any additional caching of poster (or other data), so any caching that is happening is either at the network level or, more likely, within Roku OS itself.

**Comment by ApexArray** [comment]
I was able to reproduce the issue using the steps provided.

@neilsb is correct that the caching occurs in Roku's code. From the Roku docs on [memory management]:(https://developer.roku.com/docs/developer-program/performance-guide/memory-management.md)
>Each unique image you would like to display on your channel will take up a certain amount of the limited texture memory **(however, reusing the same image on multiple nodes will not take up more memory)**

Telneting to port 8080 and running `loaded_templates` shows the cached textures. For me, the highlighted line was the image that wasn't updating.
![image](https://user-images.githubusercontent.com/91706106/209062032-73eb3270-22a1-4727-958b-ab3c8794e8cc.png)

I confirmed rebooting the Roku fixed the issue, but otherwise couldn't find a way to clear the texture cache directly. I propose one of two solutions:

**Solution 1:** Close this issue and advise users to reboot their Roku after image posters/thumbnails/backdrops. I strongly favor this option since image changes should be infrequent.

**Solution 2:** Implement cache busting, which would look something like this.
![image](https://user-images.githubusercontent.com/91706106/209063494-d725ba15-dc09-4a5c-8cd2-2d6bf58b83a4.png)

I tested this using manual static values and it worked fine. But for production, the `cacheBust` setting could simply be a timestamp that is updated when a user picks "refresh all image data" from the settings menu. This will force all images to be retrieved using unique URLs, and thus bypassing any caching.
